examples/datagen_with_al/README.md

```diff
@@ -0,0 +1,221 @@
+    --rc_weight_decay 0.01 \
+    --rc_save_total_limit 1 \
+    --rc_num_train_epochs 3 \
+    --rc_preprocessor primeqa.mrc.processors.preprocessors.squad.SQUADPreprocessor \
+    --rc_postprocessor primeqa.mrc.processors.postprocessors.squad.SQUADPostProcessor \
+    --rc_eval_metrics SQUAD \
+    --rc_metric_for_best_model f1 \
+    --rc_evaluation_strategy steps \
+    --rc_eval_steps 1000 \
+    --rc_logging_steps 1000 \
+    --rc_save_steps 1000
+```
+```bash
+# further fine-tune SQuAD-trained RC model on annotated data from BioASQ
+# select best checkpoint from previous training, i.e. checkpoint with least steps
+python run.py \
+    --train_dataset models/qg/bart-large-squad_al-bioasq-dsp-4x50/al_samples_round_3 \
+    --eval_dataset st-bioasq:validation \
+    --qg_output_dir tmp \
+    --rc_model models/rc/squad_eval-bioasq/checkpoint-8000 \
+    --rc_output_dir models/rc/squad_ft-bioasq-al-dsp-4x50 \
+    --rc_fp16 \
+    --rc_learning_rate 4e-5 \
+    --rc_do_train \
+    --rc_do_eval \
+    --rc_per_device_train_batch_size 16 \
+    --rc_per_device_eval_batch_size 128 \
+    --rc_gradient_accumulation_steps 2 \
+    --rc_warmup_ratio 0.1 \
+    --rc_weight_decay 0.01 \
+    --rc_save_total_limit 1 \
+    --rc_max_steps 500 \
+    --rc_preprocessor primeqa.mrc.processors.preprocessors.squad.SQUADPreprocessor \
+    --rc_postprocessor primeqa.mrc.processors.postprocessors.squad.SQUADPostProcessor \
+    --rc_eval_metrics SQUAD \
+    --rc_metric_for_best_model f1 \
+    --rc_evaluation_strategy steps \
+    --rc_eval_steps 10 \
+    --rc_logging_steps 10 \
+    --rc_save_steps 10
+```
+
+This can then be used for filtering:
+```bash
+python filter_gen_data.py generated_data/bioasq_bart-large-squad-al-bioasq-dsp-4x50 generated_data/bioasq_bart-large-squad-al-bioasq-dsp-4x50_rt --filter rt --rt_model models/rc/squad_ft-bioasq-al-dsp-4x50/checkpoint-220 --num_workers 5 # again, pick best checkpoint
+```
+
+
+### Train RC
+
+Train an RC model using the generated & filtered data as well as annotated data:
+
+```bash
+python run.py \
+    --train_dataset generated_data/bioasq_bart-large-squad-al-bioasq-dsp-4x50_lm models/qg/bart-large-squad_al-bioasq-dsp-4x50/al_samples_round_3 \
+    --eval_dataset st-bioasq:validation \
+    --qg_output_dir tmp \
+    --rc_model bert-base-uncased \
+    --rc_output_dir models/rc/bioasq-bart-large-squad-al-bioasq-dsp-4x50_bioasq-al-dsp-4x50_bioasq-al-dsp-4x50 \
+    --rc_fp16 \
+    --rc_learning_rate 4e-5 \
+    --rc_do_train \
+    --rc_do_eval \
+    --rc_per_device_train_batch_size 16 \
+    --rc_per_device_eval_batch_size 128 \
+    --rc_gradient_accumulation_steps 2 \
+    --rc_warmup_ratio 0.1 \
+    --rc_weight_decay 0.01 \
+    --rc_save_total_limit 1 \
+    --rc_num_train_epochs 3 \
+    --rc_preprocessor primeqa.mrc.processors.preprocessors.squad.SQUADPreprocessor \
+    --rc_postprocessor primeqa.mrc.processors.postprocessors.squad.SQUADPostProcessor \
+    --rc_eval_metrics SQUAD \
```

Comment:
What about evaluation ?

Reply:
What do you mean exactly? Evaluation is currently done during training.